sdk/core/core/src/az_span.c

```diff
@@ -36,15 +39,44 @@ AZ_NODISCARD az_span az_span_from_str(char* str)
 
 AZ_NODISCARD az_span az_span_slice(az_span span, int32_t low_index, int32_t high_index)
 {
-  AZ_PRECONDITION_RANGE(-1, high_index, az_span_capacity(span));
-  AZ_PRECONDITION(high_index == -1 || (high_index >= 0 && low_index <= high_index));
```

Comment:
This precondition is incomplete and doesn't verify that low_index is less than span.capacity when high_index == -1.

```C
az_span span = AZ_SPAN_FROM_STR("Hello world");
// This slice would return an invalid span with negative length.
az_span slice = az_span_slice(span, 1000, -1);
assert_int_equal(-989, az_span_length(slice)); // Negative length!
```